test/Generics/associated_types.swift

```diff
@@ -173,7 +173,7 @@ struct C<a : B> : B { // expected-error {{type 'C<a>' does not conform to protoc
 
 // SR-511
 protocol sr511 {
-  typealias Foo // expected-warning {{use of 'typealias' to declare associated types is deprecated; use 'associatedtype' instead}} {{3-12=associatedtype}}
+  typealias Foo // expected-error {{expected '=' in typealias declaration}}
```

Comment:
This is a case where we should provide a specialized diagnostic + Fix-It to replace "typealias" with "associatedtype"